packages/react/src/components/TreeView/TreeView-test.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+import { Document16, Folder16 } from '@carbon/icons-react';
+import { settings } from 'carbon-components';
+import TreeView, { TreeNode } from './';
+
+const { prefix } = settings;
+
+describe('TreeView', () => {
+  let wrapper;
+  let onTreeSelect;
+  let onNodeSelect;
+
+  beforeEach(() => {
+    onTreeSelect = jest.fn();
+    onNodeSelect = jest.fn();
+    wrapper = mount(
+      <TreeView label="Tree view" selected={['1']} onSelect={onTreeSelect}>
+        <TreeNode id="1" value="1" label="1" />
+        <TreeNode id="2" value="2" label="2" onSelect={onNodeSelect} />
+        <TreeNode id="5" value="5" label="5" isExpanded>
+          <TreeNode id="5-1" value="5-1" label="5-1" />
+          <TreeNode id="5-2" value="5-2" label="5-2" />
+          <TreeNode id="5-3" value="5-3" label="5-3" isExpanded>
+            <TreeNode id="5-4" value="5-4" label="5-4" />
+          </TreeNode>
+        </TreeNode>
+      </TreeView>
+    );
+  });
+
+  it('should render', () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should render with icons', () => {
+    wrapper = mount(
+      <TreeView label="Tree view" selected={['1']}>
+        <TreeNode renderIcon={Document16} id="1" value="1" label="1" />
```

Comment:
Is there any way we could have the icon supplied through children instead of with `renderIcon`? 👀

Reply:
it's possible to supply the icon through children but I chose to use `renderIcon` since it is consistent with our current component APIs elsewhere in the library, and it's also easier for me to differentiate between the node icon from the node's other children this way